group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroupMember.java

```diff
@@ -209,8 +209,19 @@ public Builder setSupportedClassicProtocols(JoinGroupRequestData.JoinGroupReques
                         .setMetadata(protocol.metadata())
                 )
             );
-            this.classicMemberMetadata = new ConsumerGroupMemberMetadataValue.ClassicMemberMetadata()
-                .setSupportedProtocols(newSupportedProtocols);
+
+            if (this.classicMemberMetadata == null) {
+                this.classicMemberMetadata = new ConsumerGroupMemberMetadataValue.ClassicMemberMetadata();
+            }
+            this.classicMemberMetadata.setSupportedProtocols(newSupportedProtocols);
+            return this;
+        }
+
+        public Builder setSessionTimeoutMs(int sessionTimeoutMs) {
```

Comment:
`setSessionTimeoutMs` is a bit misleading here in my opinion because we don't know whether it applies to the consumer of the classic protocol. I actually wonder if we should remove `setSupportedClassicProtocols` and `setSessionTimeoutMs` and use `setClassicMemberMetadata` to set both. The rational is that we always set both or none. What do you think?